eng/scripts/Get-PackageDependencies.ps1

```diff
@@ -2,12 +2,31 @@
 #Requires -Version 7.0
 
 [CmdletBinding()]
+param(
+  [switch]$Publishable,
+  [switch]$NoDev,
+  [switch]$NoReq
```

Comment:
Nit: PowerShell allows as few characters as necessary to disambiguate, but we shouldn't use shorter names that may not be obvious to everyone using it.
```suggestion
  [switch]$NoDeveloperDependencies,
  [switch]$NoRequiredDependencies
```
Let's keep them intuitive for easier use/maintenance.

Reply:
req is an output property from `cargo metadata`, short for Version Requirement
`req == '*'` means that the dependency had no version specified.
NoReq would return only entries where `req == '*'`, so NoVersion?

Reply:
or NoRequirement?

Reply:
going with NoVersion